localstack-core/localstack/services/apigateway/next_gen/provider.py

```diff
@@ -66,13 +75,35 @@ def delete_rest_api(self, context: RequestContext, rest_api_id: String, **kwargs
 
     @handler("CreateStage", expand=False)
     def create_stage(self, context: RequestContext, request: CreateStageRequest) -> Stage:
-        response = super().create_stage(context, request)
+        # TODO: we need to internalize Stages and Deployments in LocalStack, we have a lot of split logic
```

Comment:
note: most of this code is already taken from the parent class, sorry it does not make a clean diff